core/src/main/scala/kafka/log/Log.scala

```diff
@@ -47,19 +47,19 @@ import java.lang.{Long => JLong}
 import java.util.regex.Pattern
 
 object LogAppendInfo {
-  val UnknownLogAppendInfo = LogAppendInfo(-1, -1, RecordBatch.NO_TIMESTAMP, -1L, RecordBatch.NO_TIMESTAMP, -1L,
+  val UnknownLogAppendInfo = LogAppendInfo(Some(-1L), -1, RecordBatch.NO_TIMESTAMP, -1L, RecordBatch.NO_TIMESTAMP, -1L,
```

Comment:
Could we use `None` instead so that we can always guarantee that the first offset, if present, is positive? As far as I can tell, outside of test cases, we just have a couple calls to `firstOffset.get` inside `ReplicaManager` that we would need to replace with `getOrElse(-1)`. 

I would also like to remove this `UnknownLogAppendInfo` since this pattern has proven error-prone, but we can leave that for future work.